lib/puppet/functions/regsubst.rb

```diff
@@ -30,7 +30,7 @@
   #   $i3 = regsubst($ipaddress,'^(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)$','\\3')
   #   ```
   dispatch :regsubst_string do
-    param          'Variant[Array[String],String]',       :target
+    param          'Variant[Array[Variant[String,Sensitive[String]]],Variant[String,Sensitive[String]]]', :target
```

Comment:
This change allows you to call the function with an array of mixed strings and sensitive data, while I think what is implemented bellow is for this `Variant[Array[String],Sensitive[String],String]`

Reply:
You are right, this does not fit together.  I refined the Funciton and amended the Commit.  Also I added some basic Tests.